matsim/scenariogen/data/__init__.py

```diff
@@ -327,16 +332,24 @@ class Trip:
     purpose: Purpose
     sd_group: SourceDestinationGroup
     valid: bool
+    dep_district: str = ""
+    arr_district: str = ""
+    arrival: int = 0
 
 
 @dataclass
 class Activity:
     """ Activity information (including leg) """
+    # all leg information relates to the leg leading to the activity
     a_id: str
-    p_id: str
+    p_index: str
```

Comment:
This should probably not be renamed.

Reply:
Now it implies something different. Maybe an additional field is needed ?

Reply:
In the code, "p_id" sometimes meant id and sometimes index, which is ver confusing, so I renamed it, such that it is clear that it's the index.

Reply:
index might imply that it is somehow ordered and starting at 0. As you said it is used like this in some cases. But generally it is just an id which can be any string, so now index would be the wrong name.

Reply:
I see, I will make some changes then